app/src/main/java/com/iesam/psp/ut04/ex03/app/RetrofitApiClient.kt

```diff
@@ -0,0 +1,55 @@
+package com.iesam.psp.ut04.ex03.app
+
+import com.iesam.psp.ut03.ex02.domain.Failure
+import com.iesam.psp.ut04.ex03.data.remote.TapaApiModel
+import com.iesam.psp.ut04.ex03.domain.TapaModel
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.lang.Exception
+import java.util.concurrent.TimeUnit
+
+class RetrofitApiClient : ApiClient {
+
+    private val urlEndPoint: String = "https://avilaentapas.sitehub.es/api/"
+    private var apiEndPoint: ApiEndPoint
+
+    init {
+        apiEndPoint = buildApiService()
+    }
+
+    private fun buildApiService(): ApiEndPoint {
+        return build().create(ApiEndPoint::class.java)
+    }
+
+    private fun build(): Retrofit {
+        return Retrofit.Builder()
+            .baseUrl(urlEndPoint)
+            .client(buildHttpClient())
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
+
+    private fun buildHttpClient() =
+        OkHttpClient.Builder()
+            .addInterceptor(HttpLoggingInterceptor().also {
+                it.level = HttpLoggingInterceptor.Level.BODY
+            })
+            .connectTimeout(30, TimeUnit.SECONDS)
+            .build()
+
+    override fun getTapas(): Result<List<TapaApiModel>?> {
+        return try {
+            val call = apiEndPoint.getTapas()
+            val response = call.execute()
+            if (response.isSuccessful) {
+                Result.success(response.body()?.data)
+            } else {
+                Result.success(null)
```

Comment:
Vamos a intentar no devolver null, evitar trabajar con nulos. Lo que hacemos aquí es:
``
if (response.isSuccessful) {
                Result.success(response.body()?.data)
            } else {
                Result.success(null)
}
``
Aquí deberíamos devolver el listado de tapas si es que viene, este listado puede ser vacío. Una propuesta sería:

```suggestion
if (response.isSuccessful) {
    if (response.body() == null){
       return Result.success(emptyList())
   } else {
      return Result.success(response.body().data)
   }
} else {
      Result.failure(Failure.ApiError)
}
```